ocean_lib/models/test/test_side_staking.py

```diff
@@ -392,3 +387,187 @@ def test_side_staking(
 
     # Get vesting should be callable by anyone
     side_staking.get_vesting(erc20.address, another_consumer_wallet)
+
+
+@pytest.mark.unit
+def test_side_staking_steal(
+    web3, config, publisher_wallet, consumer_wallet, another_consumer_wallet
+):
+    """
+    In this test we try to steal basetoken from the pull
+    """
+    # Test initial values and setups
+
+    # How to scale the "strategy" keep the same propotion (i.e. x2, x5 x10 all the following values):
+    initial_pool_liquidity_eth = 200000
+    token_cap_eth = 1  # doesnt matter
+    join_pool_step_eth = 20000
+    datatoken_buy_amount_basetoken_eth = 5000
+    # ---------------------------------------------------------------------------------------------
+
+    initial_pool_liquidity = to_wei(initial_pool_liquidity_eth)
+    token_cap = to_wei(token_cap_eth)
+    swap_market_fee = to_wei("0.0001")
+    swap_fee = to_wei("0.0001")
+    big_allowance = to_wei("10000000000000000")
+
+    _, erc20 = deploy_erc721_erc20(
+        web3=web3,
+        config=config,
+        erc721_publisher=publisher_wallet,
+        erc20_minter=publisher_wallet,
+        cap=to_wei(to_wei("1000000000")),
+    )
+
+    erc20.mint(
+        publisher_wallet.address, to_wei("10000000000000000000"), publisher_wallet
+    )
+
+    # We use an erc20 as ocean to have unlimited balance
+    ocean_token = erc20
+
+    bpool, erc20_token, _, pool_token = create_nft_erc20_with_pool(
+        web3,
+        config,
+        publisher_wallet,
+        ocean_token,
+        swap_fee,
+        swap_market_fee,
+        initial_pool_liquidity,
+        token_cap,
+    )
+
+    ocean_token.approve(bpool.address, big_allowance, another_consumer_wallet)
+    erc20_token.approve(bpool.address, big_allowance, another_consumer_wallet)
+
+    erc20.transfer(
+        another_consumer_wallet.address,
+        erc20.balanceOf(publisher_wallet.address),
+        publisher_wallet,
+    )
+
+    # End test initial values and setups
+
+    initial_ocean_user_balance = ocean_token.balanceOf(another_consumer_wallet.address)
+    spot_prices = []
+
+    def add_spot_price():
+        spot_prices.append(
+            bpool.get_spot_price(
+                ocean_token.address, erc20_token.address, swap_market_fee
+            )
+        )
+
+    add_spot_price()
```

Comment:
Can you add some comments to each of these steps explaining what the test is trying to do?